src/plugins/data/public/search/search_interceptor/search_interceptor.ts

```diff
@@ -445,14 +451,66 @@ export class SearchInterceptor {
     if (this.bFetchDisabled) {
       const { executionContext, strategy, ...searchOptions } = this.getSerializableOptions(options);
       return this.deps.http
-        .post(`/internal/search/${strategy}${request.id ? `/${request.id}` : ''}`, {
-          version: '1',
-          signal: abortSignal,
-          context: executionContext,
-          body: JSON.stringify({
-            ...request,
-            ...searchOptions,
-          }),
+        .post<IKibanaSearchResponse>(
+          `/internal/search/${strategy}${request.id ? `/${request.id}` : ''}`,
+          {
+            version: '1',
+            signal: abortSignal,
+            context: executionContext,
+            body: JSON.stringify({
+              ...request,
+              ...searchOptions,
+              stream:
+                strategy === ESQL_ASYNC_SEARCH_STRATEGY ||
+                strategy === ENHANCED_ES_SEARCH_STRATEGY ||
+                strategy === undefined, // undefined strategy is treated as enhanced ES
+            }),
+            asResponse: true,
+          }
+        )
+        .then((rawResponse) => {
+          const response = rawResponse.body!;
+          const warning = rawResponse.response?.headers.get('warning');
+          const requestParams =
+            response.requestParams ??
+            JSON.parse(rawResponse.response?.headers.get('requestParams') || '{}');
+          const error = (response as unknown as Record<string, unknown>).error;
+          if (error) {
+            // eslint-disable-next-line no-throw-literal
+            throw {
+              attributes: {
+                error,
+                rawResponse: response,
+                requestParams,
+              },
+            };
+          }
+          switch (strategy) {
+            case ENHANCED_ES_SEARCH_STRATEGY:
+              if (response.rawResponse) return response;
+              const typedResponse = response as unknown as AsyncSearchGetResponse;
+              const shimmedResponse = shimHitsTotal(typedResponse.response);
+              return {
+                id: typedResponse.id,
+                isPartial: typedResponse.is_partial,
+                isRunning: typedResponse.is_running,
+                rawResponse: shimmedResponse,
+                warning,
+                requestParams,
+                ...getTotalLoaded(shimmedResponse),
+              };
+            case ESQL_ASYNC_SEARCH_STRATEGY:
+              const esqlResponse = response as unknown as SqlGetAsyncResponse;
```

Comment:
Should be `EqlEqlSearchResponseBase`

Reply:
? can you link me to this type

Reply:
I was mistaken, it doesn't look like the JS client has the correct types for this class yet.